tests/Opus/Doi/DataCiteXmlGeneratorTest.php

```diff
@@ -139,4 +139,171 @@ public function testGetStylesheetPathConfigured()
+        $doc->addFile($file);
+
+        $generator = new Opus_Doi_DataCiteXmlGenerator();
+        $result = $generator->getXml($doc);
+
+        $this->assertNotContains('<sizes><size>0 KB</size></sizes>', $result);
+        $this->assertNotContains('<formats><format>pdf</format></formats>', $result);
+    }
+
+    /**
+     * Creates a txt-file with random size.
+     *
+     * @return string path of file
+     * @throws Zend_Exception
+     */
+    private function createTestFile() {
+        $config = Zend_Registry::get('Zend_Config');
+        $path = $config->workspacePath . DIRECTORY_SEPARATOR . uniqid();
+
+        $src_path = $path . DIRECTORY_SEPARATOR . 'src';
+        mkdir($src_path, 0777, true);
+
+        $filename_nonzero = $src_path . DIRECTORY_SEPARATOR . 'foobar-nonzero.txt';
+        $fh = fopen($filename_nonzero, 'w');
+
+        if ($fh == false) {
+            $this->fail("Unable to write file $filename_nonzero.");
+        }
+
+        $rand = rand(1000, 100000);
+        for ($i = 0; $i < $rand; $i++) {
+            fwrite($fh, ".");
+        }
+
+        fclose($fh);
+
+        return $filename_nonzero;
+    }
+
+    /**
+     * The DataCite-XML should not contain files, which are invisible in oai
+     * Test if both visible files are shown
+     */
+    public function testFileInformationVisible()
+    {
+        $doc = new Opus_Document();
+        $this->addRequiredPropsToDoc($doc);
+
+        $filename = $this->createTestFile();
+
+        $file = New Opus_File();
+        $file->setVisibleInOai(1);
+        $file->setTempFile($filename);
+        $file->setPathName('copied-foobar-nonzero.txt');
+        $doc->addFile($file);
+        $doc->store();
+
+        $filename2 = $this->createTestFile();
+
+        $file2 = New Opus_File();
+        $file2->setVisibleInOai(1);
+        $file2->setTempFile($filename2);
+        $file2->setPathName('copied-foobar-nonzero_2.txt');
+        $doc->addFile($file2);
+        $doc->store();
+
+        $generator = new Opus_Doi_DataCiteXmlGenerator();
+        $result = $generator->getXml($doc);
+
+        $size = intval(round($file->getFileSize()/1024));
+        $size2 = intval(round($file2->getFileSize()/1024));
+
+        $this->assertContains("<sizes><size>$size KB</size><size>$size2 KB</size></sizes>", $result);
+        $this->assertContains('<formats><format>text/plain</format><format>text/plain</format></formats>', $result);
+    }
+
+    /**
+     * The DataCite-XML should not contain files, which are invisible in oai
+     * Test if visible file is shown and invisible file is hided
+     */
+    public function testMixedFileInformationVisibility()
+    {
+        $doc = new Opus_Document();
+        $this->addRequiredPropsToDoc($doc);
+
+        $filename = $this->createTestFile();
+
+        $file = New Opus_File();
+        $file->setVisibleInOai(0);
+        $file->setTempFile($filename);
+        $file->setPathName('copied-foobar-nonzero3.txt');
```

Comment:
Nicht kritisch, aber Konsistenz im Code ist gut. Mal "nonzero3", dann "nonzero_4". Letztendlich sollte jeder dieser Tests die gleichen Dateinamen verwenden können, weil die Testdateien nach jedem Test bereinigt werden.

Reply:
Das Problem habe ich auch gesehen. Bei mir findet beim ausführen dieser Klasse keine Bereinigung statt. Es findet sich auch kein TearDown in dieser Klasse. Da die Datein nicht gelöscht wurden, kam es zu Fehlern beim ausführen. "Datei schon vorhanden"
Daher diese Lösung. Man könnte hier vielleicht einen TearDown einbauen. Man muss nur schauen, ob es keine anderen Probleme auslöst.